sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosTableSchemaInferrer.scala

```diff
@@ -175,4 +178,48 @@ private object CosmosTableSchemaInferrer
   private def inferDataTypeFromArrayNode(node: ArrayNode, allowNullForInferredProperties: Boolean): Option[DataType] = {
     Option(node.get(0)).map(firstElement => inferDataTypeFromJsonNode(firstElement, allowNullForInferredProperties))
   }
+
+  /**
+   * It looks for the most compatible type between two given DataTypes.
+   * i.e.: {{{
+   *   val dataType1 = IntegerType
+   *   val dataType2 = DoubleType
+   *   assert(compatibleType(dataType1,dataType2)==DoubleType)
+   * }}}
+   *
+   * @param t1 First DataType to compare
+   * @param t2 Second DataType to compare
+   * @return Compatible type for both t1 and t2
+   */
+  private def compatibleType(t1: DataType, t2: DataType): DataType = {
+    //TypeCoercion.findTightestCommonTypeOfTwo(t1, t2) match {
+    TypeCoercion.findTightestCommonType(t1, t2) match {
+      case Some(commonType) => commonType
+
+      case None =>
+        // t1 or t2 is a StructType, ArrayType, or an unexpected type.
+        (t1, t2) match {
+          case (other: DataType, NullType) => other
+          case (NullType, other: DataType) => other
```

Comment:
I wonder if we can add some test cases for all branches here including null type

Reply:
The test was already there before `it should "map duplicate properties when one is nullable"`

Reply:
I will add to it on the other direction (first node having null, second having a value) which should take care of the other case

Reply:
Added another test for it.